R/xml_find.R

```diff
@@ -93,6 +108,6 @@ xml_find_one.xml_nodeset <- function(x, xpath, ns = character()) {
   if (length(x) == 0)
     return(xml_nodeset())
 
-  nodes <- lapply(x, function(x) node_find_one(x$node, x$doc, xpath = xpath, nsMap = ns))
-  xml_nodeset(lapply(nodes, function(y) xml_node(y, x[[1]]$doc)))
+  xml_nodeset(lapply(x, function(x)
```

Comment:
I think the checking for duplicate nodes needs to happen in `xml_nodeset()` because even for `xml_find_one` could return duplicate nodes (e.g. if your xpath was `.//x[1]`)